presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -398,19 +399,26 @@ public synchronized void updatePartitionStatistics(String databaseName, String t
     }
 
     @Override
-    public synchronized List<String> getAllViews(String databaseName)
+    public synchronized List<String> getTablesWithParameter(String databaseName, String parameterKey, String parameterValue)
     {
+        requireNonNull(parameterKey, "parameterKey is null");
+        requireNonNull(parameterValue, "parameterValue is null");
+
         List<String> tables = getAllTables(databaseName);
 
-        List<String> views = tables.stream()
+        return tables.stream()
                 .map(tableName -> getTable(databaseName, tableName))
                 .filter(Optional::isPresent)
                 .map(Optional::get)
-                .filter(table -> table.getTableType().equals(VIRTUAL_VIEW.name()))
+                .filter(table -> parameterValue.equals(table.getParameters().get(parameterKey)))
                 .map(Table::getTableName)
                 .collect(toImmutableList());
+    }
 
-        return views;
+    @Override
+    public synchronized List<String> getAllViews(String databaseName)
+    {
+        return getTablesWithParameter(databaseName, PRESTO_VIEW_FLAG, "true");
```

Comment:
@electrum please review the change here.

Reply:
Looks good